crates/rpc-client/src/client.rs

```diff
@@ -86,7 +86,7 @@ impl<T> RpcClient<T> {
     ///
     /// Note: This will only set the poll interval for the client if it is the only reference to the
     /// inner client. If the reference is held by many, then it will not update the poll interval.
-    pub fn with_poll_interval(self, poll_interval: u64) -> Self {
+    pub fn with_poll_interval(self, poll_interval: Duration) -> Self {
```

Comment:
@yash-atreya I seem to remember us specifically avoiding changing this to `Duration` in another PR. Do you remember why?

Reply:
I believe I'm thinking of #437 , which chose to represent duration as atomic internally to allow sharing between threads and setting via the `Arc<_>`. because we still represent as atomic internally, I'm fine with this api change, although it feels weird to do the conversion both ways